TuningApp.py

```diff
@@ -0,0 +1,592 @@
+#!/usr/bin/python3
+import os
+import os.path
+import shutil
+import sys
+import warnings
+from concurrent.futures import ThreadPoolExecutor
+
+import cv2
+import numpy as np
+from picamera2 import Picamera2, Platform
+from picamera2.previews.qt import QGlPicamera2
+from PyQt5.QtCore import QTimer, pyqtSignal
+from PyQt5.QtWidgets import (QApplication, QHBoxLayout, QLabel, QLineEdit,
+                             QListWidget, QListWidgetItem, QMainWindow,
+                             QMessageBox, QPushButton, QTabWidget, QVBoxLayout,
+                             QWidget)
+
+# finding ctt folder
+ctt_directory = os.path.join(os.path.expanduser("~"), 'libcamera/utils/raspberrypi/ctt')
+sys.path.insert(1, ctt_directory)
+from ctt import Camera, run_ctt
+from ctt_image_load import dng_load_image
+from ctt_macbeth_locator import find_macbeth, get_macbeth_chart
+
+
+# doesnt let runtime warning print
+def fxn():
+    warnings.warn("runtime", RuntimeWarning)
+
+
+# Configuring the camera
+# sorting functions for camera modes
+def sortFn(d):
+    return d['crop_limits'][-1] * d['crop_limits'][-2]
+
+
+def sortBitDepth(d):
+    return d['bit_depth']
+
+
+picam2 = Picamera2()
+candidate_modes = []
+
+
+for x in picam2.sensor_modes:
```

Comment:
I'd still prefer `mode` instead of `x`...

Reply:
fixed